### PR TITLE
[quant][fx] Ensure qconfig works for QAT with multiple modules

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63343

Summary:
The previous implementation had a bug where we were trying to modify an ordered dict value while iterating through it.
This fixes it by creating a copy before modifying it.

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_qconfig_qat_module_type

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30346116](https://our.internmc.facebook.com/intern/diff/D30346116)

### CODE REVIEW

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30346116).

File: torch/quantization/fx/prepare.py [hunk 1]
```diff
@@ -163,7 +163,8 @@ def update_qconfig_for_qat(
     all_qat_mappings = get_combined_dict(
         get_default_qat_module_mappings(), additional_qat_module_mapping)
     object_type_dict = qconfig_dict.get("object_type", None)
-    for k, v in object_type_dict.items():
+    new_object_type_dict = object_type_dict.copy()
```
Comment (raghuramank100) [inline torch/quantization/fx/prepare.py:166]: Why is the copy needed here?

**Review by raghuramank100 — APPROVED** [review]

File: torch/quantization/fx/prepare.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/fx/prepare.py:166]: re Raghu: this is for avoid iterating through the dict while changing it I think

@supriyar looks like `object_type_dict` can be None according to L165, wondering why that does not fail when we call None.items()

Comment (supriyar) [reply torch/quantization/fx/prepare.py:166]: I checked this, on line 165, if it is None then `object_type_dict` gets assigned an empty `OrderedDict()` so items returns empty.

**Pull request closed without merging**